python/cuml/cuml/ensemble/randomforest_common.pyx

```diff
@@ -545,3 +598,57 @@ class BaseRandomForestModel(Base, InteropMixin):
                 align_bytes=align_bytes,
             )
         return fil_model
+
+    @property
+    def oob_score_(self):
+        """Out-of-bag score for the training dataset.
+
+        Only available if oob_score=True.
+
+        Returns
+        -------
+        score : float
+            Score of the training dataset using an out-of-bag estimate.
+            For classification, this is the accuracy score.
+            For regression, this is the R-squared score.
+        """
+        if not self.oob_score:
+            raise AttributeError(
+                "oob_score_ is only available when oob_score=True"
+            )
+        if self._oob_score_ < 0:
+            raise NotFittedError(
```

Comment:
Similar to the above comment, in scikit-learn a `AttributeError` is raised if you access a fitted attribute before fitting the model. Can we copy that behaviour here?

(Same comment about the behaviour in `def feature_importances_`)

Reply:
Actually I have removed the property descriptors and straight away access class members. This way I think we are deferring the error handling to python without writing custom text ourselves.